src/com/test/optional/Task/TaskService.java

```diff
@@ -0,0 +1,71 @@
+package com.test.optional.Task;
+import com.test.optional.Task.domain.Task;
+import com.test.optional.Task.domain.User;
+//https://github.com/shekhargulati/java8-the-missing-tutorial/
+//https://github.com/wytten/java8-optional/blob/master/src/test/java/optional/PersonTest.java
+//https://medium.com/faun/working-on-null-elegantly-with-java-optional-62f5e65869c5
+import java.time.LocalDate;
+import java.util.Optional;
+
+class TaskService {
+    private TaskRepository taskRepository;
+
+    TaskService(TaskRepository taskRepository){
+         this.taskRepository = taskRepository;
+    }
+
+    // orElseThrow: Method explained
+    String taskTitle(String taskId) {
+        return taskRepository.
+                find(taskId).
+                map(Task::getTitle).
+                orElseThrow(() -> new TaskNotFoundException(String.format("No task exist for id '%s'",taskId)));
+    }
+
+    // orElse Method: Task Assigned to with flatMap
+    String taskAssignedTo(String taskId) {
+        return taskRepository.
+                find(taskId).
+                flatMap(task -> task.getAssignedTo().map(User::getUsername)).
+                orElse("Task Not Assigned yet");
+    }
+
+    // Task assigned to with Map
+    String taskAssignedToWithMap(String taskId) {
+        Optional<Optional<String>> assignedOptionalWrapper = taskRepository.find(taskId).map(task -> task.getAssignedTo().map(User::getUsername));
```

Comment:
Optional of Optional?

Reply:
Search for map vs flatMap for this problem

Reply:
Yeah, here I am just explaining the difference between flatMap & map. Here I have created two methods to achieve the same task, one is with a map & other is using a flat map just to show the difference between map & flat map